analytics/lib/counts.py

```diff
@@ -304,14 +307,20 @@ def do_increment_logging_stat(
 
     table = stat.data_collector.output_table
     if table == RealmCount:
-        assert isinstance(zerver_object, Realm)
-        id_args: Dict[str, Union[Realm, UserProfile, Stream]] = {"realm": zerver_object}
+        assert isinstance(zulip_object, Realm)
+        id_args: Dict[str, Optional[Union[Realm, UserProfile, Stream, "RemoteZulipServer"]]] = {
+            "realm": zulip_object
+        }
     elif table == UserCount:
-        assert isinstance(zerver_object, UserProfile)
-        id_args = {"realm": zerver_object.realm, "user": zerver_object}
-    else:  # StreamCount
-        assert isinstance(zerver_object, Stream)
-        id_args = {"realm": zerver_object.realm, "stream": zerver_object}
+        assert isinstance(zulip_object, UserProfile)
+        id_args = {"realm": zulip_object.realm, "user": zulip_object}
+    elif table == StreamCount:
+        assert isinstance(zulip_object, Stream)
+        id_args = {"realm": zulip_object.realm, "stream": zulip_object}
+    else:
+        assert table == RemoteInstallationCount
+        assert isinstance(zulip_object, RemoteZulipServer)
+        id_args = {"server": zulip_object, "remote_id": None}
```

Comment:
Same comment here that I think it'd be better to have an `else: raise AssertionError` structure at the end.